File: japiwdbe/WdbeQUsrList.java
/**
  * \file WdbeQUsrList.h
  * Java API code for record of table TblWdbeQUsrList
  * \author Alexander Wirthmueller
  * \date created: 11 Jul 2020
  * \date modified: 11 Jul 2020
  */

package apiwdbe;

import org.w3c.dom.*;
import sbecore.*;

public class WdbeQUsrList {
	
	public WdbeQUsrList(
				int jnum
				, String stubGrp
				, String stubOwn
				, String stubRefWdbeMPerson
				, String sref
				, String stubRefWdbeMUsergroup
				, String srefIxVState
				, String titIxVState
				, String srefIxWdbeVLocale
				, String titIxWdbeVLocale
				, String srefIxWdbeVUserlevel
				, String titIxWdbeVUserlevel
			) {
		this.jnum = jnum;
		this.stubGrp = stubGrp;
		this.stubOwn = stubOwn;
		this.stubRefWdbeMPerson = stubRefWdbeMPerson;
		this.sref = sref;
		this.stubRefWdbeMUsergroup = stubRefWdbeMUsergroup;
		this.srefIxVState = srefIxVState;
		this.titIxVState = titIxVState;
		this.srefIxWdbeVLocale = srefIxWdbeVLocale;
		this.titIxWdbeVLocale = titIxWdbeVLocale;
		this.srefIxWdbeVUserlevel = srefIxWdbeVUserlevel;
		this.titIxWdbeVUserlevel = titIxWdbeVUserlevel;
	};

	public int jnum;
	public String stubGrp;
	public String stubOwn;
	public String stubRefWdbeMPerson;
	public String sref;
	public String stubRefWdbeMUsergroup;
	public String srefIxVState;
	public String titIxVState;
	public String srefIxWdbeVLocale;
	public String titIxWdbeVLocale;
	public String srefIxWdbeVUserlevel;
	public String titIxWdbeVUserlevel;
	
	public boolean readXML(
				Document doc
				, String basexpath
				, boolean addbasetag
			) {
		if (addbasetag) basexpath = Xmlio.checkUclcXPaths(doc, basexpath, "WdbeQUsrList");

		if (Xmlio.checkXPath(doc, basexpath)) {
			stubGrp = Xmlio.extractStringUclc(doc, basexpath, "stubGrp", "grp", null, 0);
			stubOwn = Xmlio.extractStringUclc(doc, basexpath, "stubOwn", "own", null, 0);
			stubRefWdbeMPerson = Xmlio.extractStringUclc(doc, basexpath, "stubRefWdbeMPerson", "prs", null, 0);
			sref = Xmlio.extractStringUclc(doc, basexpath, "sref", "srf", null, 0);
			stubRefWdbeMUsergroup = Xmlio.extractStringUclc(doc, basexpath, "stubRefWdbeMUsergroup", "usg", null, 0);
			srefIxVState = Xmlio.extractStringUclc(doc, basexpath, "srefIxVState", "ste", null, 0);
			titIxVState = Xmlio.extractStringUclc(doc, basexpath, "titIxVState", "ste2", null, 0);
			srefIxWdbeVLocale = Xmlio.extractStringUclc(doc, basexpath, "srefIxWdbeVLocale", "lcl", null, 0);
			titIxWdbeVLocale = Xmlio.extractStringUclc(doc, basexpath, "titIxWdbeVLocale", "lcl2", null, 0);
			srefIxWdbeVUserlevel = Xmlio.extractStringUclc(doc, basexpath, "srefIxWdbeVUserlevel", "ulv", null, 0);
			titIxWdbeVUserlevel = Xmlio.extractStringUclc(doc, basexpath, "titIxWdbeVUserlevel", "ulv2", null, 0);
			
			return true;
		};

		return false;
	};
};

File: apiwdbe/WdbeQUsrList.cpp
/**
	* \file WdbeQUsrList.cpp
	* API code for table TblWdbeQUsrList (implementation)
	* \copyright (C) 2016-2020 MPSI Technologies GmbH
	* \author Alexander Wirthmueller (auto-generation)
	* \date created: 5 Dec 2020
  */
// IP header --- ABOVE

#include "WdbeQUsrList.h"

using namespace std;
using namespace Sbecore;
using namespace Xmlio;

/******************************************************************************
 class WdbeQUsrList
 ******************************************************************************/

WdbeQUsrList::WdbeQUsrList(
			const uint jnum
			, const string stubGrp
			, const string stubOwn
			, const string stubRefWdbeMPerson
			, const string sref
			, const string stubRefWdbeMUsergroup
			, const string srefIxVState
			, const string titIxVState
			, const string srefIxWdbeVLocale
			, const string titIxWdbeVLocale
			, const string srefIxWdbeVUserlevel
			, const string titIxWdbeVUserlevel
		) {
	this->jnum = jnum;
	this->stubGrp = stubGrp;
	this->stubOwn = stubOwn;
	this->stubRefWdbeMPerson = stubRefWdbeMPerson;
	this->sref = sref;
	this->stubRefWdbeMUsergroup = stubRefWdbeMUsergroup;
	this->srefIxVState = srefIxVState;
	this->titIxVState = titIxVState;
	this->srefIxWdbeVLocale = srefIxWdbeVLocale;
	this->titIxWdbeVLocale = titIxWdbeVLocale;
	this->srefIxWdbeVUserlevel = srefIxWdbeVUserlevel;
	this->titIxWdbeVUserlevel = titIxWdbeVUserlevel;
};

bool WdbeQUsrList::readXML(
			xmlXPathContext* docctx
			, string basexpath
			, bool addbasetag
		) {
	bool basefound;

	if (addbasetag)
		basefound = checkUclcXPaths(docctx, basexpath, basexpath, "WdbeQUsrList");
	else
		basefound = checkXPath(docctx, basexpath);

	if (basefound) {
		extractStringUclc(docctx, basexpath, "stubGrp", "grp", stubGrp);
		extractStringUclc(docctx, basexpath, "stubOwn", "own", stubOwn);
		extractStringUclc(docctx, basexpath, "stubRefWdbeMPerson", "prs", stubRefWdbeMPerson);
		extractStringUclc(docctx, basexpath, "sref", "srf", sref);
		extractStringUclc(docctx, basexpath, "stubRefWdbeMUsergroup", "usg", stubRefWdbeMUsergroup);
		extractStringUclc(docctx, basexpath, "srefIxVState", "ste", srefIxVState);
		extractStringUclc(docctx, basexpath, "titIxVState", "ste2", titIxVState);
		extractStringUclc(docctx, basexpath, "srefIxWdbeVLocale", "lcl", srefIxWdbeVLocale);
		extractStringUclc(docctx, basexpath, "titIxWdbeVLocale", "lcl2", titIxWdbeVLocale);
		extractStringUclc(docctx, basexpath, "srefIxWdbeVUserlevel", "ulv", srefIxWdbeVUserlevel);
		extractStringUclc(docctx, basexpath, "titIxWdbeVUserlevel", "ulv2", titIxWdbeVUserlevel);
	};

	return basefound;
};

/******************************************************************************
 class ListWdbeQUsrList
 ******************************************************************************/

ListWdbeQUsrList::ListWdbeQUsrList() {
};

ListWdbeQUsrList::ListWdbeQUsrList(
			const ListWdbeQUsrList& src
		) {
	nodes.resize(src.nodes.size(), NULL);
	for (unsigned int i = 0; i < nodes.size(); i++) nodes[i] = new WdbeQUsrList(*(src.nodes[i]));
};

ListWdbeQUsrList& ListWdbeQUsrList::operator=(
			const ListWdbeQUsrList& src
		) {
	nodes.resize(src.nodes.size(), NULL);
	for (unsigned int i = 0; i < nodes.size(); i++) nodes[i] = new WdbeQUsrList(*(src.nodes[i]));

	return *this;
};

ListWdbeQUsrList::~ListWdbeQUsrList() {
	clear();
};

void ListWdbeQUsrList::clear() {
	for (unsigned int i = 0; i < nodes.size(); i++) if (nodes[i]) delete nodes[i];
	nodes.resize(0);
};

bool ListWdbeQUsrList::readXML(
			xmlXPathContext* docctx
			, string basexpath
			, bool addbasetag
		) {
	bool basefound;

	vector<unsigned int> ics;
	vector<bool> shorttags;

	WdbeQUsrList* rec = NULL;

	string s;

	if (addbasetag)
		basefound = checkUclcXPaths(docctx, basexpath, basexpath, "ListWdbeQUsrList");
	else
		basefound = checkXPath(docctx, basexpath);

	extractList(docctx, basexpath, "WdbeQUsrList", "row", "jnum", ics, shorttags);

	clear();

	for (unsigned int i = 0; i < ics.size(); i++) {
		rec = new WdbeQUsrList(ics[i]);

		if (shorttags[i]) s = basexpath + "/row[@jnum='" + to_string(ics[i]) + "']";
		else s = basexpath + "/WdbeQUsrList[@jnum='" + to_string(ics[i]) + "']";

		if (rec->readXML(docctx, s)) nodes.push_back(rec);
		else delete rec;
	};

	return basefound;
};

